include/chopper/count/count_kmers.hpp

```diff
@@ -62,28 +55,33 @@ inline void count_kmers(robin_hood::unordered_map<std::string, std::vector<std::
     #pragma omp parallel for schedule(static) num_threads(config.num_threads)
     for (size_t i = 0; i < cluster_vector.size(); ++i)
     {
+        chopper::sketch::hyperloglog sketch(config.sketch_bits);
+
+        auto hash_seq_and_add_to_sketch = [&sketch] (auto & seq, auto && compute_fn)
+        {
+            for (auto && hash : seq | compute_fn)
+                sketch.add(reinterpret_cast<char*>(&hash), sizeof(hash));
+        };
+
         // read files
-        std::vector<std::vector<seqan3::dna4>> sequence_vector{};
         for (auto const & filename : cluster_vector[i].second)
+        {
             for (auto && [seq] : sequence_file_type{filename})
-                sequence_vector.push_back(seq);
-
-        chopper::sketch::hyperloglog sketch(config.sketch_bits);
-
-        if (config.disable_minimizers)
-            for (auto && seq : sequence_vector)
-                add_hashes_to_sketch(seq, compute_kmers, sketch);
-        else
-            for (auto && seq : sequence_vector)
-                add_hashes_to_sketch(seq, compute_minimiser, sketch);
+            {
+                if (config.disable_minimizers)
+                    hash_seq_and_add_to_sketch(seq, compute_kmers);
+                else
+                    hash_seq_and_add_to_sketch(seq, compute_minimiser);
+            }
+        }
```

Comment:
Can you either:
* make the lambda open the file, i.e. put the inner for-loop into the lambda
* re-arrange this block such that we do not have an if check for every sequence in every cluster

Reply:
very good point